src/main/java/com/jfrog/ide/common/configuration/JfrogCliDriver.java

```diff
@@ -90,4 +105,57 @@ public CommandResults runCommand(File workingDirectory, String[] args, List<Stri
 
         return commandResults;
     }
+
+    public void downloadCliIfNeeded(String destinationPath, String jfrogCliVersion) throws IOException {
+        // verify installed cli version
+        String cliVersion = extractVersion(version(null));
+        log.debug("Local CLI version is: " + cliVersion);
+        // cli is installed but not the correct version
+        if (cliVersion != null && cliVersion.equals(jfrogCliVersion)) {
+            log.info("Local Jfrog CLI version has been verified and is compatible. Proceeding with its usage.");
+        } else {
+            log.info(String.format("JFrog CLI is either not installed or the current version is incompatible. " +
+                    "Initiating download of version %s to the destination: %s.", jfrogCliVersion, destinationPath));
+            downloadCliFromReleases(jfrogCliVersion, destinationPath);
+        }
+    }
+
+    public void downloadCliFromReleases(String cliVersion, String destinationFolder) throws IOException {
+        String[] urlParts = {"jfrog-cli/v2-jf", cliVersion, "jfrog-cli-" + getOSAndArc(), jfrogExec};
+        String fileLocationInReleases = String.join("/", urlParts);
+        Path basePath = Paths.get(destinationFolder);
+        String destinationPath = basePath.resolve(jfrogExec).toString();
+
+        // download executable from releases and save it in 'destinationPath'
+        try{
+            ServerConfig serverConfig = getServerConfig();
+            ArtifactoryManagerBuilder artifactoryManagerBuilder = createAnonymousAccessArtifactoryManagerBuilder(JFROG_CLI_RELEASES_URL, serverConfig.getProxyConfForTargetUrl(JFROG_CLI_RELEASES_URL), log);
+            ArtifactoryManager artifactoryManager = artifactoryManagerBuilder.build();
+            File cliExecutable = artifactoryManager.downloadToFile(fileLocationInReleases, destinationPath);
+            // setting the file as executable
+            if(!cliExecutable.setExecutable(true)){
+                log.error(String.format("Failed to set downloaded CLI as executable. Path: %s", destinationPath));
+            }
+            log.debug(String.format("Downloaded CLI to %s. Permission te execute: %s", destinationPath, cliExecutable.canExecute()));
```

Comment:
```suggestion
            if(!cliExecutable.setExecutable(true)){
                log.error(String.format("Failed to set downloaded CLI as executable. Path: %s", destinationPath));
            } else {
                log.debug(String.format("Downloaded CLI to %s. Permission te execute: %s", destinationPath, cliExecutable.canExecute()));
            }
```
if it fails to set, it will print both logs